c10/core/TensorImpl.cpp

```diff
@@ -770,7 +772,7 @@ void TensorImpl::Extend(int64_t num, float growthPct) {
   SizesVector newDims(sizes_and_strides.begin(), sizes_and_strides.end());
   newDims[0] += num;
   if (!storage_.data()) {
-    Resize(newDims);
+    Resize(std::move(newDims));
```

Comment:
Cannot do this, newDims is used afterwards. For safety let's not touch the legacy Caffe2 operations

Reply:
Huh? How is it used afterwards? newDims is a local variable and the next line is a void return statement and therefore the dtor on newDims is immediately called. I'll make this change anyway though. newDims is used later, but not on this if branch.

Reply:
oh oops, I missed the return, my bad!